dev-packages/cli/src/download-plugins.ts

```diff
@@ -114,13 +135,17 @@ export default async function downloadPlugins(options: DownloadPluginsOptions =
             const client = new OVSXClient({ apiVersion, apiUrl });
             // De-duplicate extension ids to only download each once:
             const ids = new Set<string>(Array.from(extensionPacks.values()).flat());
-            await Promise.all(Array.from(ids, async id => {
+            const extensionPacksToDownload = await Promise.all(Array.from(ids, async id => {
                 const extension = await client.getLatestCompatibleExtensionVersion(id);
+                const version = extension?.version;
```

Comment:
It looks like this code is still making some network calls in parallel to determine the latest compatible version of the given plugin. Would you like to deparallelize that, as well?

Reply:
I have not seen any issues with the ovsx client api requests... so I dont have any reason to run those in serial...

Reply:
By now, I suspect that the problematic stage is indeed where compatible versions are determined. The `includeAllVersions` flag probably leads to a significant response size, multiplied by about 80 requests in parallel.

Reply:
Interesting fact: The API response sends an `extensions` array whose entries each have an `allVersions` map. That is, response size grows asymptotically with the square of the number of versions.

Reply:
The median version count currently seems to be at 457. The query for `vscode.bat` comes with 487 versions, and the JSON response is 20,377,701 bytes. With line separators (as prettified by `jq .`), 236680 out of 258585 lines contain version-like crossref URLs (characterized by containing exactly 6 slashes). That's 92% of lines and 96% by byte count after prettifying.

Reply:
This approach does not scale. I'd suggest omitting the `allVersions` maps from the `extensions` entries if `includeAllVersions` is true. But that's an issue for Open-VSX. Here I suppose we should simply resort to query versions one extension at a time (with `--no-parallel` at least).

Reply:
Actually, since the `extensions` array lists most recent versions first, and the logic of `getLatestCompatibleExtensionVersion` is to pick the first compatible one, and if you have, and use, a streamable JSON decoder, one optimization would be to cancel the version query download once a suitable `extensions` array entry has been found...

Reply:
Trying to get the total byte count of the query responses for the bundled extensions. Might take an hour or so.

Reply:
@ccorn I believe that the recent comments are adding unecessary noise to the pull-request for the author, if you want to investigate further optimizations you can comment instead on the issue. I think the optimizations you recently commented on are likely out-of-scope for the pull-request at this time. Implement parallel downloads should be the full scope, modifications to how we handle the `allVersions` would be out-of-scope. I don't think we should perform multiple open-vsx calls per extension to determine the compatible version.